src/python/CMSSpark/reports/aggregate_phedex.py

```diff
@@ -0,0 +1,135 @@
+        return 0
+    return 1
+
+def unix2human(tstamp):
+    "Convert unix time stamp into human readable format"
+    return time.strftime('%Y%m%d', time.gmtime(tstamp))
+
+def quiet_logs(sc):
+    """
+    Sets logger's level to ERROR so INFO logs would not show up.
+    """
+    logger = sc._jvm.org.apache.log4j
+    logger.LogManager.getRootLogger().setLevel(logger.Level.ERROR)
+
+def run(date, fout, yarn=None, verbose=None):
+    """
+    Main function to run pyspark job. It requires a schema file, an HDFS directory
+    with data and optional script with mapper/reducer functions.
+    """
+    # define spark context, it's main object which allow to communicate with spark
+    ctx = spark_context('cms', yarn, verbose)
+
+    quiet_logs(ctx)
+    
+    sqlContext = HiveContext(ctx)
+
+    fromdate = '%s-%s-%s' % (date[:4], date[4:6], date[6:])
+    todate = fromdate
+    # read Phedex tables
+    tables = {}
+    tables.update(phedex_tables(sqlContext, verbose=verbose, fromdate=fromdate, todate=todate))
+    phedex_df = tables['phedex_df']
+
+    # register user defined function
+    unix2date = udf(unix2human, StringType())
+    siteFilter = udf(site_filter, IntegerType())
+
+    # aggregate phedex info into dataframe
+    cols = ['node_name', 'dataset_name', 'block_bytes', 'replica_time_create', 'br_user_group_id']
+    pdf = phedex_df.select(cols).where(siteFilter(col('node_name')) == 1)\
+            .groupBy(['node_name', 'dataset_name', 'replica_time_create', 'br_user_group_id'])\
+            .agg({'block_bytes':'sum'})\
+            .withColumn('date', lit(date))\
+            .withColumn('replica_date', unix2date(col('replica_time_create')))\
+            .withColumnRenamed('sum(block_bytes)', 'size')\
+            .withColumnRenamed('dataset_name', 'dataset')\
+            .withColumnRenamed('node_name', 'site')\
+            .withColumnRenamed('br_user_group_id', 'groupid')
+    pdf.registerTempTable('pdf')
+    pdf.persist(StorageLevel.MEMORY_AND_DISK)
+
+    # write out results back to HDFS, the fout parameter defines area on HDFS
+    # it is either absolute path or area under /user/USERNAME
+    if  fout:
+        cols = ['date','site','dataset','size','replica_date', 'groupid']
+        # don't write header since when we'll read back the data it will
+        # mismatch the data types, i.e. headers are string and rows
+        # may be different data types
+        pdf.select(cols)\
+            .write.format("com.databricks.spark.csv")\
+            .option("header", "true").save(fout)
+
+    ctx.stop()
+
+def main():
+    "Main function"
+    opts = get_options()
+    print("Input arguments: %s" % opts)
+    time0 = time.time()
+    fout = opts.fout
+    date = opts.date
+    verbose = opts.verbose
+    yarn = opts.yarn
+
+    run(date, fout, yarn, verbose)
+
+    print('Start time  : %s' % time.strftime('%Y-%m-%d %H:%M:%S GMT', time.gmtime(time0)))
```

Comment:
the same comment as in aggregate_campaign_tier.py

Reply:
Valentin, I didn't use options parser from conf.py because arguments list is different in each file. For example some files require --limit argument and others don't. And in conf.py file --no-log4j argument is defined which I never use. Do you want me to replace my option parsing code with the class defined in conf.py by adding --limit argument to conf.py and making proper use of --no-log4j in my scripts?

Reply:
Oh, I see that OptionsParser exposes ArgumentParser instance so I can add any additional arguments in each script, so that's good. But what to do with --no-log4j argument? I can see that in some of your scripts this argument is not used as well. Do you think that it's a good idea to have an argument in script that is not used?